network-mux/src/Network/Mux/Bearer/Socket.hs

```diff
@@ -112,17 +111,4 @@ socketAsMuxBearer tracer sd = do
           return ts
 
       sduSize :: IO Word16
-#if defined(mingw32_HOST_OS)
-      sduSize = return 13000 -- MaxSegment isn't supported on Windows
-#else
-      sduSize = do
-          -- XXX it is really not acceptable to call getSocketOption for every SDU we want to send
-          {- getSocketOption for MaxSegment is not supported for AF_UNIX sockets -}
-          addr <- Socket.getSocketName sd
-          case addr of
-               Socket.SockAddrUnix _ -> return 0xffff
-               _                     -> do
-                    mss <- Socket.getSocketOption sd Socket.MaxSegment
-                    -- 1260 = IPv6 min MTU minus TCP header, 8 = mux header size
-                    return $ fromIntegral $ max (1260 - 8) (min 0xffff (15 * mss - 8))
-#endif
+      sduSize = return 12288
```

Comment:
We don't need to wrap it in `IO` unless we plan to have dynamic sdu size in the future.

Reply:
True, will change it.